Delete tdabasinskas/backstage provider

tdabasinskas opened the pull request:
This provider was superseded by [`datolabs-io/backstage`](https://search.opentofu.org/provider/datolabs-io/backstage/latest).

ghost commented:
Hi @tdabasinskas thank you for this PR. Generally we prefer to add a warning to the provider so users of it won't be met with a confusing error message but are instead directed to the correct place. See [this example](https://github.com/opentofu/registry/blob/beac92753ddad1eb16adb2d743bc9c2d761465a9/providers/h/hashicorp/incapsula.json#L3). Would this be ok with you?

tdabasinskas committed:
feat(tdabasinskas): add deprecation warning for backstage provider

Added a warning message to inform users that the `tdabasinskas/backstage` provider is deprecated and they should use `datolabs-io/backstage` instead. This helps guide users to the maintained version of the provider.

Signed-off-by: Tomas Dabašinskas <tomas@dabasinskas.net>

tdabasinskas commented:
Yeah, makes sense - updated.
